core/domain/role_services.py

```diff
@@ -122,211 +123,142 @@
-    feconf.ROLE_ID_GUEST: [],
-    feconf.ROLE_ID_LEARNER: [feconf.ROLE_ID_GUEST],
-    feconf.ROLE_ID_MODERATOR: [feconf.ROLE_ID_TOPIC_MANAGER],
-    feconf.ROLE_ID_TOPIC_MANAGER: [feconf.ROLE_ID_COLLECTION_EDITOR]
-}
 
-# This dict represents the unique actions that belong to a particular role.
-# Unique in the sense that the action belongs to this role but can't be
-# inherited from any other role.
-#   key -> name of role
-#   value -> list of unique actions.
-#
-# NOTE FOR DEVELOPERS :
-# - Follow the Playbook in wiki (https://github.com/oppia/oppia/wiki/
-#   Instructions-for-editing-roles-or-actions) before making any changes to
-#   this dict.
-ROLE_ACTIONS = {
-    feconf.ROLE_ID_ADMIN: [
-        ACTION_ACCEPT_ANY_SUGGESTION,
-        ACTION_ACCEPT_ANY_VOICEOVER_APPLICATION,
-        ACTION_CHANGE_STORY_STATUS,
-        ACTION_CHANGE_TOPIC_STATUS,
-        ACTION_CREATE_NEW_SKILL,
-        ACTION_CREATE_NEW_TOPIC,
-        ACTION_DELETE_ANY_ACTIVITY,
-        ACTION_DELETE_ANY_SKILL,
-        ACTION_DELETE_TOPIC,
-        ACTION_EDIT_ANY_ACTIVITY,
-        ACTION_EDIT_ANY_STORY,
-        ACTION_EDIT_ANY_TOPIC,
-        ACTION_EDIT_SKILLS,
-        ACTION_EDIT_SKILL_DESCRIPTION,
-        ACTION_MANAGE_EMAIL_DASHBOARD,
-        ACTION_MANAGE_TOPIC_RIGHTS,
-        ACTION_MODIFY_ROLES_FOR_ANY_ACTIVITY,
-        ACTION_PUBLISH_ANY_ACTIVITY,
-        ACTION_PUBLISH_OWNED_SKILL
-    ],
-    feconf.ROLE_ID_BANNED_USER: [
-    ],
-    feconf.ROLE_ID_COLLECTION_EDITOR: [
-        ACTION_CREATE_COLLECTION,
-    ],
-    feconf.ROLE_ID_EXPLORATION_EDITOR: [
-        ACTION_ACCESS_CREATOR_DASHBOARD,
-        ACTION_CREATE_EXPLORATION,
-        ACTION_DELETE_OWNED_PRIVATE_ACTIVITY,
-        ACTION_EDIT_OWNED_ACTIVITY,
-        ACTION_SUBSCRIBE_TO_USERS,
-        ACTION_MANAGE_ACCOUNT,
-        ACTION_MODIFY_ROLES_FOR_OWNED_ACTIVITY,
-        ACTION_PUBLISH_OWNED_ACTIVITY,
-        ACTION_RATE_ANY_PUBLIC_EXPLORATION,
-        ACTION_SUGGEST_CHANGES,
-        ACTION_SUBMIT_VOICEOVER_APPLICATION,
-    ],
-    feconf.ROLE_ID_GUEST: [
-        ACTION_PLAY_ANY_PUBLIC_ACTIVITY,
-    ],
-    feconf.ROLE_ID_LEARNER: [
-        ACTION_FLAG_EXPLORATION,
-        ACTION_ACCESS_LEARNER_DASHBOARD,
-    ],
-    feconf.ROLE_ID_MODERATOR: [
-        ACTION_ACCESS_MODERATOR_PAGE,
-        ACTION_DELETE_ANY_PUBLIC_ACTIVITY,
-        ACTION_EDIT_ANY_PUBLIC_ACTIVITY,
-        ACTION_PLAY_ANY_PRIVATE_ACTIVITY,
-        ACTION_SEND_MODERATOR_EMAILS,
-        ACTION_UNPUBLISH_ANY_PUBLIC_ACTIVITY,
-    ],
-    feconf.ROLE_ID_TOPIC_MANAGER: [
-        ACTION_ACCESS_TOPICS_AND_SKILLS_DASHBOARD,
-        ACTION_DELETE_ANY_QUESTION,
-        ACTION_EDIT_ANY_QUESTION,
-        ACTION_EDIT_OWNED_STORY,
-        ACTION_EDIT_OWNED_TOPIC,
-        ACTION_EDIT_SKILLS,
-        ACTION_EDIT_ANY_SUBTOPIC_PAGE,
-        ACTION_MANAGE_QUESTION_SKILL_STATUS,
-        ACTION_VISIT_ANY_QUESTION_EDITOR,
-        ACTION_VISIT_ANY_TOPIC_EDITOR
-    ]
+def _get_unique_actions_list(*actions):
```

Comment:
Ah, I see. Yes, one suggestion is to just expand each list for now or update the definitions to the new versions (where some permissions are removed). But, if you want to keep it as-is, I think it's fine; just add a TODO here to remove this function once all lists have been revamped.

Reply:
I have added a TODO comment to remove the function. [Reason: I want to edit existing roles and actions in a separate PR as I wanted to do it after checking all edge cases to avoid creating issues for end-users.]